tool/tsh/config.go

```diff
@@ -121,3 +160,39 @@ func onConfig(cf *CLIConf) error {
 	fmt.Print(sb.String())
 	return nil
 }
+
+func onConfigProxy(cf *CLIConf) error {
+	proxyHost, proxyPort, err := net.SplitHostPort(cf.Proxy)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	targetHost, targetPort, err := net.SplitHostPort(cf.ConfigProxyTarget)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// If the node is suffixed by either the root or leaf cluster name, remove it.
+	targetHost = strings.TrimSuffix(targetHost, "."+proxyHost)
+	targetHost = strings.TrimSuffix(targetHost, "."+cf.ConfigProxyLeaf)
+
+	args := []string{
+		"-p",
+		proxyPort,
+		proxyHost,
+		"-s",
+	}
+
+	if cf.ConfigProxyLeaf != "" {
+		args = append(args, fmt.Sprintf("proxy:%s:%s@%s", targetHost, targetPort, cf.ConfigProxyLeaf))
+	} else {
+		args = append(args, fmt.Sprintf("proxy:%s:%s", targetHost, targetPort))
+	}
```

Comment:
I think we can always include the cluster name, even when it's root
You should be able to remove some conditional branches this way

Reply:
That's much nicer, done!